File: sum using recursion/solution1.py
arr = [1,2,3,4,5,6,7,8,9,10]
size = len(arr)


def sol(arr,size):
    if size == 0:
        return 0
    return arr[len(arr)-size] + sol(arr, size-1)

print(sol(arr, size))

File: sum using recursion/solution1.cpp
#include <bits/stdc++.h>
using namespace std;

int SuM(int arr[],int len) {
    if (len == 0) {
        return 0;
    }
    return arr[0] + SuM(arr+1,len-1);
}

int main() {
    int arr[] = {1,2,3,4,5};
    int len = sizeof(arr) / sizeof(arr[0]);
    cout << SuM(arr,len);
}
